widget/button.go

```diff
@@ -105,18 +105,27 @@ func alignedPosition(align ButtonAlign, padding, objectSize, layoutSize fyne.Siz
 // applyTheme updates this button to match the current theme
 func (b *buttonRenderer) applyTheme() {
 	b.label.TextSize = theme.TextSize()
-	b.label.Color = theme.TextColor()
+	c := theme.TextColor()
+	if b.button.Style == PrimaryButton {
+		c = theme.PrimaryTextColor()
+	}
+
 	if b.button.Disabled() {
-		b.label.Color = theme.DisabledTextColor()
+		c = theme.DisabledTextColor()
 	}
+	b.label.Color = c
 }
 
 func (b *buttonRenderer) BackgroundColor() color.Color {
 	switch {
 	case b.button.Disabled():
 		return theme.DisabledButtonColor()
 	case b.button.Style == PrimaryButton:
-		return theme.PrimaryColor()
+		c := theme.PrimaryColor()
+		if b.button.hovered {
+			c = theme.Brighten(c, 1)
```

Comment:
Shouldn't this brighten in light mode, and darken in dark mode, or visa versa?

Reply:
added code to detect if the current theme is light/dark based on the predominant BG color, and then brighten/dim accordingly

Is subtle, but it looks pretty cool